domain/fhir/eoverdracht/service.go

```diff
@@ -0,0 +1,298 @@
+
+	composition, sections, patient, err := s.ResolveComposition(ctx, fhirCompositionPath)
+	if err != nil {
+		return NursingHandoff{}, fmt.Errorf("could not get nursing handoff composition: %w", err)
+	}
+	nursingHandoff := NursingHandoff{
+		Composition: *composition,
+		Patient:     *patient,
+	}
+
+	nursingDiagnosisEntries, ok := sections[CarePlanCode+"/"+NursingDiagnosisCode]
+	if !ok {
+		return NursingHandoff{}, errors.New("NursingDiagnosis not set in composition")
+	}
+
+	// the nursing diagnosis contains both conditions and procedures
+	for _, entry := range nursingDiagnosisEntries {
+		if condition, ok := entry.(resources.Condition); ok {
+			nursingHandoff.Problems = append(nursingHandoff.Problems, condition)
+		}
+		if procedure, ok := entry.(fhir.Procedure); ok {
+			nursingHandoff.Interventions = append(nursingHandoff.Interventions, procedure)
+		}
+	}
+
+	return nursingHandoff, nil
+}
+
+func (s transferService) findTaskInputOutputByCode(ios []resources.TaskInputOutput, code datatypes.Code) *resources.TaskInputOutput {
+	for _, io := range ios {
+		if fhir.FromCodePtr(io.Type.Coding[0].Code) == string(code) {
+			return &io
+		}
+	}
+	return nil
+}
+
+func (s transferService) ResolveComposition(ctx context.Context, compositionPath string) (*fhir.Composition, map[string][]interface{}, *resources.Patient, error) {
+	composition := fhir.Composition{}
+	patient := resources.Patient{}
+	sections := map[string][]interface{}{}
+
+	err := s.fhirClient.ReadOne(ctx, "/"+compositionPath, &composition)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("error while fetching the composition(id=%s): %w", compositionPath, err)
+	}
+
+	// Fetch the Patient
+	err = s.fhirClient.ReadOne(ctx, "/"+fhir.FromStringPtr(composition.Subject.Reference), &patient)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("error while fetching the transfer subject (patient): %w", err)
+	}
+
+	// Fill a map with sections by code
+	for _, l1 := range composition.Section {
+		l1code := fhir.FromCodePtr(l1.Code.Coding[0].Code)
+		if l1code == "" {
+			continue
+		}
+		for _, l2 := range l1.Section {
+			l2code := fhir.FromCodePtr(l2.Code.Coding[0].Code)
+			if l2code == "" {
+				continue
+			}
+			var entryResources []interface{}
+			for _, l2entry := range l2.Entry {
+				entryPath := fhir.FromStringPtr(l2entry.Reference)
+				resourceTypeStr := strings.Split(entryPath, "/")[0]
+				switch resourceTypeStr {
+				case "Condition":
+					resource := resources.Condition{}
+					err := s.fhirClient.ReadOne(ctx, entryPath, &resource)
+					if err != nil {
+						return nil, nil, nil, err
+					}
+					entryResources = append(entryResources, resource)
+				case "Procedure":
+					resource := fhir.Procedure{}
+					err := s.fhirClient.ReadOne(ctx, entryPath, &resource)
+					if err != nil {
+						return nil, nil, nil, err
+					}
+					entryResources = append(entryResources, resource)
+				}
+
+			}
+			sections[l1code+"/"+l2code] = entryResources
+		}
+	}
```

Comment:
I get what it's supposed to do but the code quality isn't great. If we only support conditions and procedures doesn't it make sense to create a structure which holds the patient, conditions and procedures in a struct? Then we wouldn't have to deal with interfaces and checking their type.